jcl/src/java.base/share/classes/java/lang/Object.java

```diff
@@ -294,6 +294,21 @@ public final void wait(long time, int frac) throws InterruptedException {
 			Blocker.end(blocking);
 		}
 	}
+/*[ELSE] JAVA_SPEC_VERSION < 24 */
+	if ((time < 0) || (frac < 0)) {
+		throw new IllegalArgumentException("timeout value is negative");
```

Comment:
I think we may already throw these exceptions internally

Reply:
we do, I added this to match RI's bytecode as it would be a faster path to avoid having to call into native.